filesystem/fat32/directoryentry.go

```diff
@@ -413,11 +416,47 @@ func convertLfnSfn(name string) (string, string, bool, bool) {
 		isLFN = true
 	}
 
+	truncate := func(name string, n int) string {
+		return name[0:6] + "~" + strconv.Itoa(n)
+	}
+
+	hash := func(shortName string, n int) string {
+		h := fnv.New32a()
+		// the input name may have more information than the short name
+		h.Write([]byte(name))
+		return shortName[0:2] + fmt.Sprintf("%04X", (h.Sum32()%0x10000)) + "~" + strconv.Itoa(n)
+	}
+
+	random := func() string {
+		return fmt.Sprintf("%06X", (rand.Int63()%0x1000000)) + "~" + strconv.Itoa(1+rand.Intn(8))
+	}
```

Comment:
`random` is a bit too generic of a name, and probably will confuse. We should pull this out too, but make it more specifically-named.